model/textparse/protobufparse.go

```diff
@@ -422,6 +423,16 @@ func (p *ProtobufParser) Next() (Entry, error) {
 		default:
 			return EntryInvalid, fmt.Errorf("unknown metric type for metric %q: %s", name, p.mf.GetType())
 		}
+		unit := p.mf.GetUnit()
+		if len(unit) > 0 {
+			sfx := fmt.Sprintf("_%s", unit)
+			if p.mf.GetType() == dto.MetricType_COUNTER {
```

Comment:
```suggestion
			if p.mf.GetType() == dto.MetricType_COUNTER && strings.HasSuffix(name, "_total") {
```

In the classic Prometheus format, there might be counters with a name not ending on `_total`. (It's just a recommendation, not enforced like in OpenMetrics.)

(I thought I had commented on this, but apparently I haven't saved my comment in the 1st round of review.)